DOCSP-30350 Java Write Exception Troubleshooting (#710)
mongodb/docs-java: source/crud/insert.txt

      insertOne() document id: BsonObjectId{value=...}
      insertMany() document ids: {0=BsonObjectId{value=...}, 1=BsonObjectId{value=...}}

.. _java-insert-troubleshooting:

Troubleshooting
---------------

Write Exception
~~~~~~~~~~~~~~~

The driver throws a `MongoWriteException
<{+core-api+}/MongoWriteException.html>`__ for any write errors that occur when
performing single write operations. A ``MongoWriteException`` object has an
``error`` field containing the ``WriteError`` object that caused it.

> Consider a collection with a rule where the value of the ``quantity`` field must

Reviewer: ditto

      insertOne() document id: BsonObjectId{value=...}
      insertMany() document ids: {0=BsonObjectId{value=...}, 1=BsonObjectId{value=...}}

.. _java-insert-troubleshooting:

Troubleshooting
---------------

> .. include:: /includes/crud/write-error.rst
> 
> Write Exception
> ~~~~~~~~~~~~~~~
> 
> The driver creates a write exception when it creates a ``WriteError`` object. The
> driver throws a `MongoWriteException <{+core-api+}/MongoWriteException.html>`__
> for any write errors that occur when performing single write operations.
> 
> A ``MongoWriteException`` object has an ``error`` field containing the
> ``WriteError`` object that caused it.

Reviewer 2: Same suggestion as in the bulk write page. Included some tweaks for brevity.

```suggestion
Write Exception
~~~~~~~~~~~~~~~

The
driver throws a `MongoWriteException <{+core-api+}/MongoWriteException.html>`__
for any write errors that occur when performing single write operations. The ``error`` field of a ``MongoWriteException`` contains a ``WriteError`` object that you can access to determine the cause of the error.
```

> Example
> ```````
> 
> For example, the driver throws a ``MongoWriteException`` if you
> attempt to insert a document into a collection that violates the collection's
> schema validation rules. Suppose the collection has a rule where the value of
> the ``quantity`` field must be an ``int`` type. If you attempt to insert a
> document where the value of ``quantity`` is ``"three"``, the driver prints the
> following error message:

Reviewer 2: Similar suggestion as above.

```suggestion
Consider a collection with a rule where the value of
the ``quantity`` field must be of the ``int`` type. The driver throws the following example error message if you attempt to insert a document where the value of the ``quantity`` field is ``"three"``.
```
